### DISCUSSION TITLE
why Fields aren't Livewire components?

### DISCUSSION THREAD

**Discussion by AbanoubNassem** [filamentphp/filament#4763] [General]
why `Fields` aren't `Livewire` components?

I find this is as a limitation , especially if I'm developing a `field-plugin`, that depends on a 3rd party javascript library.

I would like to access the parent form using `$this->getFormLivewire()` , and still treat the field as a `Livewire` component , so I can manipulate it individually . 

is there any reason behind that decision?

**Comment by danharrin** [comment]
If you've worked with Livewire before, you'll know that many nested Livewire components on a page really slow things down. Also, it's currently not possible to easily bind state between child and parent components. You can access the Livewire component inside the form field on render using `$this->getLivewire()` or `$this` in the view.

**Reply by AbanoubNassem** [reply]
well, I know that I can access the form-livewire using `$this->getLivewire()` , it was confusing at the beginning , because, I thought I was accessing the `field-livewire` , not the form , that's why I said , it should be `$this->getFormLivewire()` , to make it more clear.

but still the problem with manipulating the field individually is the main issue, I have tried using the form-events and listen to it inside the form-field , but there was a weird behaviour  , if I send an event from the `Field-View` , and listen to it inside the `Field` class , it reset the whole form state , so for example , if I'm using a 3d party javascript plugin , updated the front-end , then send an `event` to the `Form` , then the `Form` pass it down to the `Field` , I find it , re-renders the whole form , which make the 3d party javascript plugin, resets and loses its state , that's why I was trying to access the `Field` individually .

Hopefully I made it more clear now.

**Reply by danharrin** [reply]
You can always embed a Livewire component inside the field view if you want, but it's really not required for 99% of fields

**Reply by AbanoubNassem** [reply]
alright , thanks for clearing things up, also I have published a new plugin , please review it , when you have time.
